lib/html-proofer/runner.rb

```diff
@@ -167,9 +167,7 @@ def print_failed_tests
       sorted_failures = SortedIssues.new(@failures, @options[:error_sort], @logger)
 
       sorted_failures.sort_and_report
-      count = @failures.length
-      failure_text = pluralize(count, 'failure', 'failures')
-      raise @logger.colorize :red, "HTML-Proofer found #{failure_text}!"
+      return @failures
```

Comment:
Sorry, but you are removing the existing use case of printing errors on the command line for the sake of your specific use case of iterating over the failures in a script. 

An option should be passed to enable the returning of these failures, or at least some TTY detection to note that when this is run from the terminal, the logging is the option.